src/client.cpp

```diff
@@ -109,9 +111,10 @@ ADDON_STATUS ADDON_Create(void* hdl, void* props)
 
   Logger::Log(LogLevel::LEVEL_INFO, "starting PVR client");
 
-  ADDON_ReadSettings();
+  tvheadend::Settings *set = new tvheadend::Settings();
+  set->ReadSettings();
 
-  tvh = new CTvheadend(reinterpret_cast<PVR_PROPERTIES *>(props));
+  tvh = new CTvheadend(reinterpret_cast<PVR_PROPERTIES *>(props), set);
```

Comment:
As long as you still have a settings singleton (global variable `settings`) it does make NO sence to hold it additionally as a member of `CTvheadend`.